src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.OSX.cs

```diff
@@ -268,28 +276,12 @@ public static SecurityStatusPal Renegotiate(ref SafeFreeCredentials? credentials
                 SecurityStatusPal status = PerformHandshake(sslHandle);
                 if (status.ErrorCode == SecurityStatusPalErrorCode.CredentialsNeeded)
                 {
-                    // we should not be here if CertSelectionDelegate is null but better check before dereferencing..
-                    if (sslAuthenticationOptions.CertSelectionDelegate != null)
+                    X509Certificate2? clientCertificate = secureChannel.SelectClientCertificate(out _);
+                    if (clientCertificate != null && clientCertificate.HasPrivateKey)
```

Comment:
It may be worth of adding trace log if the certificate does not have key. Perhaps move this change to SelectClientCertificate and make it common.

Reply:
Acutally, `SelectClientCertificate` Ensures that the returned cert has private key (it otherwise returns null). Trace logging such cases is already present.

Which means we can remove the PrivateKey check